gremlin-dotnet/test/Gremlin.Net.UnitTest/Driver/JsonMessageSerializerTests.cs

```diff
@@ -0,0 +1,66 @@
+#region License
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#endregion
+
+using Gremlin.Net.Driver;
+using Gremlin.Net.Driver.Messages;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Text;
+using Xunit;
+
+namespace Gremlin.Net.UnitTest.Driver
+{
+    // Will be used in future to switch to new .NET Core Json deserializer
+    public class JsonMessageSerializerTests
```

Comment:
I'm not sure about the value these tests add as they are basically just asserting how a JSON deserializer should treat edge cases, but OK, they could become useful when we try out the new .NET JSON serializer, so I'm also not against adding them.
Could you however please refactor the tests a bit if you want to add them to Gremlin.Net? Each test should only test a single case, should have a clear name that reflects what the expected behaviour is and it should ideally be split into the three phases _arrange_, _act_, and _assert_, separated by newlines (see for example [here](http://wiki.c2.com/?ArrangeActAssert) for a reference).
The [GremlinServerTests](https://github.com/apache/tinkerpop/blob/master/gremlin-dotnet/test/Gremlin.Net.UnitTest/Driver/GremlinServerTests.cs#L29) also show this nicely.